articles/ssh-in-nodejs/index.md

```diff
@@ -0,0 +1,149 @@
+
+### What is SSH?
+[SSH](https://www.ssh.com/ssh/) stands for **Secure Shell** (also called **Secure Socket Shell**) and is a protocol that allows users to access a remote system or a virtual machine using the command line or the terminal. SSH establishes a shell session with the remote host, and the user will be able to manage the remote host by executing commands on their local computer via a terminal or command-line interface. In other words, users can "log in" to a  virtual machine and manipulate it based on their requirements. SSH is a very powerful tool for IT professionals and system administrators who deal with configuring and managing servers on a daily basis. Without SSH, users would have to physically go to remote data centers and connect to the computers, which is not feasible or efficient. 
+
+SSH is based on the [client-server](https://www.britannica.com/technology/client-server-architecture) model, where a client is an application on the local system, and the server is the host you wish to connect to. The client application takes the remote host information, such as username, password, etc., as input and establishes an encrypted SSH session with the remote host. In the server, a [daemon](https://www.ssh.com/ssh/sshd/) process constantly listens to a port for any incoming SSH requests and checks to see if the credentials sent by the client are correct. If not, the request is rejected, and an SSH connection is not established. 
+
+#### Using SSH
+Most UNIX based operating systems such as Ubuntu, MAC OS, etc. come with an SSH client pre-installed. To establish a connection, open the terminal and execute the following command:
+
+`ssh <username>@<hostname or host ip>`
+
+The username field specifies the user id of the machine you are connecting to. The hostname is the name of the remote system and can be found by typing `hostname` in the terminal. Users can also specify the host IP instead of the hostname. for example:
+
+`ssh adith@219.123.101.201`
+
+This establishes a connection with the host whose IP is: 219.123.101.201 and the user is adith. 
+
+### SSH in NodeJs
+We are going to build an application that connects to a virtual machine and extracts information on the filesystem health of the VM. The application monitors the disk utilization in the VM and flags the partitions that have a disk utilization greater than 90%. This is useful for system administrators as it tells which partitions need to be monitored and increased in size before an actual failure of the system due to low space. 
+
+Node has an array of libraries for establishing an SSH connection. In this article, we are going to use [simple-ssh](https://www.npmjs.com/package/simple-ssh). Simple-ssh is a robust and easy to use library that allows us to connect to a VM and execute Linux commands using Node.
+
+Install the library by running the following command:
+
+`npm install simple-ssh`
```

Comment:
Someone who hasn't used Node.js probably won't understand this. You don't necessarily need to write a full tutorial on installing Node, but a link to one would be useful.